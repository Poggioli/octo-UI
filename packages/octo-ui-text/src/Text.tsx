import { forwardRef } from "@octo-ui/core";
import { StyledText } from "./styles";
import { TextProps } from "./types";

const Text = forwardRef<typeof StyledText, TextProps>(
  ({ color, ...props }, forwardedRef) => {
    const css = {
      color: color ?? "$grey12",
      ...props.css,
    };

    return <StyledText {...props} ref={forwardedRef} css={css} />;
  }
);
Text.toString = () => `.${StyledText.className}`;

export { StyledText };
export { Text };
