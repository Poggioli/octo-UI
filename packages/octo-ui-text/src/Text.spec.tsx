import { render } from "@testing-library/react";
import { Text } from "./Text";

describe("<Text />", () => {
  it(`GIVEN a <Text />
      WHEN size props is not setted
      THEN should render a <Text /> using size 3`, () => {
    const rendered = render(<Text>Text</Text>);
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Text />
      WHEN toString is called
      THEN return a string`, () => {
    expect(Text.toString()).toBe(".octo-UI-c-gWerhV");
  });

  describe("Size props", () => {
    [1, 2, 3, 4, 5, 6, 7, 8, 9].forEach((size) => {
      it(`GIVEN a <Text /> with size props
        WHEN size props is ${size}
        THEN should render a <Text /> using size ${size}`, () => {
        const rendered = render(<Text size={size as any}>Text</Text>);
        expect(rendered.container).toMatchSnapshot();
      });
    });
  });

  describe("color props", () => {
    it(`GIVEN a <Text />
        WHEN color props is $grey11
        THEN should render a <Text /> $grey11 as color`, () => {
      const rendered = render(<Text color="$grey11">Text</Text>);
      expect(rendered.container).toMatchSnapshot();
    });
  });
});
