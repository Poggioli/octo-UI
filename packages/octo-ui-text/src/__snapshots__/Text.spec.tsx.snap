// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> GIVEN a <Text />
      WHEN size props is not setted
      THEN should render a <Text /> using size 3 1`] = `
<div>
  <span
    class="octo-UI-c-gWerhV octo-UI-c-gWerhV-iMVRpC-size-3 octo-UI-c-gWerhV-iisyIHw-css"
  >
    Text
  </span>
</div>
`;

exports[`<Text /> Size props GIVEN a <Text /> with size props
        WHEN size props is 1
        THEN should render a <Text /> using size 1 1`] = `
<div>
  <span
    class="octo-UI-c-gWerhV octo-UI-c-gWerhV-EDrvg-size-1 octo-UI-c-gWerhV-iisyIHw-css"
  >
    Text
  </span>
</div>
`;

exports[`<Text /> Size props GIVEN a <Text /> with size props
        WHEN size props is 2
        THEN should render a <Text /> using size 2 1`] = `
<div>
  <span
    class="octo-UI-c-gWerhV octo-UI-c-gWerhV-cBLpOj-size-2 octo-UI-c-gWerhV-iisyIHw-css"
  >
    Text
  </span>
</div>
`;

exports[`<Text /> Size props GIVEN a <Text /> with size props
        WHEN size props is 3
        THEN should render a <Text /> using size 3 1`] = `
<div>
  <span
    class="octo-UI-c-gWerhV octo-UI-c-gWerhV-iMVRpC-size-3 octo-UI-c-gWerhV-iisyIHw-css"
  >
    Text
  </span>
</div>
`;

exports[`<Text /> Size props GIVEN a <Text /> with size props
        WHEN size props is 4
        THEN should render a <Text /> using size 4 1`] = `
<div>
  <span
    class="octo-UI-c-gWerhV octo-UI-c-gWerhV-gHgbiV-size-4 octo-UI-c-gWerhV-iisyIHw-css"
  >
    Text
  </span>
</div>
`;

exports[`<Text /> Size props GIVEN a <Text /> with size props
        WHEN size props is 5
        THEN should render a <Text /> using size 5 1`] = `
<div>
  <span
    class="octo-UI-c-gWerhV octo-UI-c-gWerhV-cNGvkq-size-5 octo-UI-c-gWerhV-iisyIHw-css"
  >
    Text
  </span>
</div>
`;

exports[`<Text /> Size props GIVEN a <Text /> with size props
        WHEN size props is 6
        THEN should render a <Text /> using size 6 1`] = `
<div>
  <span
    class="octo-UI-c-gWerhV octo-UI-c-gWerhV-cMAlre-size-6 octo-UI-c-gWerhV-iisyIHw-css"
  >
    Text
  </span>
</div>
`;

exports[`<Text /> Size props GIVEN a <Text /> with size props
        WHEN size props is 7
        THEN should render a <Text /> using size 7 1`] = `
<div>
  <span
    class="octo-UI-c-gWerhV octo-UI-c-gWerhV-cGJhQn-size-7 octo-UI-c-gWerhV-iisyIHw-css"
  >
    Text
  </span>
</div>
`;

exports[`<Text /> Size props GIVEN a <Text /> with size props
        WHEN size props is 8
        THEN should render a <Text /> using size 8 1`] = `
<div>
  <span
    class="octo-UI-c-gWerhV octo-UI-c-gWerhV-cpZglh-size-8 octo-UI-c-gWerhV-iisyIHw-css"
  >
    Text
  </span>
</div>
`;

exports[`<Text /> Size props GIVEN a <Text /> with size props
        WHEN size props is 9
        THEN should render a <Text /> using size 9 1`] = `
<div>
  <span
    class="octo-UI-c-gWerhV octo-UI-c-gWerhV-hlJriB-size-9 octo-UI-c-gWerhV-iisyIHw-css"
  >
    Text
  </span>
</div>
`;

exports[`<Text /> color props GIVEN a <Text />
        WHEN color props is $grey11
        THEN should render a <Text /> $grey11 as color 1`] = `
<div>
  <span
    class="octo-UI-c-gWerhV octo-UI-c-gWerhV-iMVRpC-size-3 octo-UI-c-gWerhV-ibuyHBD-css"
  >
    Text
  </span>
</div>
`;
